app/controllers/sections_controller.rb

```diff
@@ -8,7 +8,7 @@ def new
   end
   
   def show
-    @section = Section.find(params[:id])
+    @section = find_section(params[:id])
```

Comment:
If an instructor isn't in a section, they should still be able to see it. This would fail tho because find_section would consider their action invalid. Separate the finding of a section for actions like this, and ones where instructor permissions are necessary. A general function like is_instructor_in _section should probably be made in application controller to be used all around the application (like sections, projects, and teams.